### PR TITLE
dist/tools/codespell: add interactive mode

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

This PR allows calling the codespell script with `CODESPELL_INTERACTIVE=1 dist/tools/codespell/check.sh`, which will then interactively go through the found typos and directly change fixed files.

### CODE REVIEW

File: dist/tools/codespell/check.sh [hunk 1]
```diff
@@ -41,15 +41,23 @@ CODESPELL_OPTS+=" --check-hidden"
...
-if [ -n "${ERRORS}" ]
-then
-    printf "%sThere are typos in the following files:%s\n\n" "${CERROR}" "${CRESET}"
-    printf "%s\n" "${ERRORS}"
-    # TODO: return 1 when all typos are fixed
-    exit 0
+if [ "${CODESPELL_INTERACTIVE}" = "1" ]; then
+    # interactive mode
+    CODESPELL_OPTS+=" -w -i3"
+    exec ${CODESPELL_CMD} ${CODESPELL_OPTS} ${FILES}
```
Comment (aabadie) [inline dist/tools/codespell/check.sh]: How about the following ?

```suggestion
    exec ${CODESPELL_CMD} ${CODESPELL_OPTS} ${FILES}
    exit 0
fi
```

This way no need to change the non interactive mode code.

Comment (kaspar030) [reply dist/tools/codespell/check.sh]: exec replaces the current shell, lines after are not evaluated anymore:

```
[kaspar@ng riot (add_codespell_interactive_mode)]$ cat test.sh 
#!/bin/sh

exec /bin/echo FOO
echo bar
[kaspar@ng riot (add_codespell_interactive_mode)]$ sh test.sh 
FOO
[kaspar@ng riot (add_codespell_interactive_mode)]$
```

Comment (aabadie) [reply dist/tools/codespell/check.sh]: Then remove `exec` keyword from my suggestion ?

Comment (kaspar030) [reply dist/tools/codespell/check.sh]: What's your aim? 

> This way no need to change the non interactive mode code.

You mean, not move into the else block?

Comment (kaspar030) [reply dist/tools/codespell/check.sh]: ping @aabadie you mean to refactor this so the original code is not in an else case?

Comment (aabadie) [reply dist/tools/codespell/check.sh]: > you mean to refactor this so the original code is not in an else case?

Yes

**Comment by aabadie** [comment]
@kaspar030, please squash!

**Comment by miri64** [comment]
Re-queue sanctioned by @kaspar030 offline. This was done to have this PR benefit from the speed-boost in #13337.

**Pull request merged**